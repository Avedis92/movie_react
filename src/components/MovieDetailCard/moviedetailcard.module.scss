@import "../sasscommonmixxins/flex.scss";
@import "../sasscommonmixxins/mediaqueries.scss";

.container {
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
  .InnerContainer {
    background-color: rgba(0, 0, 0, 0.486);
    .Infos {
      @include flexElements(flex, row, center, space-between);
      padding-top: 10%;
      padding-left: 5%;
      padding-bottom: 4%;
      flex-wrap: wrap;
      .posterContainer {
        img {
          height: 400px;
          border-radius: 10px;
          @include mobile {
            height: 350px;
          }
        }
        @include mobile {
          margin-left: 18%;
        }
      }
      .movieInfoContainer {
        width: 75%;
        .movieHeaders {
          color: white;
          p {
            font-size: 20px;
            @include mobile {
              font-size: 16px;
            }
          }
          h1 {
            margin-bottom: 2%;
            @include mobile {
              font-size: 20px;
            }
          }
          h2 {
            margin-bottom: 2%;
            @include mobile {
              font-size: 18px;
            }
          }
          .movieGenres {
            margin-bottom: 1%;
            h2 {
              display: inline;
              @include mobile {
                font-size: 18px;
              }
            }
          }
        }
        .movieOverview {
          color: white;
          font-size: 18px;
          @include mobile {
            font-size: 16px;
            width: 100vw;
          }
        }
        .trailerWatch {
          width: 10%;
          height: 50px;
          :nth-child(1) {
            width: 100%;
          }
          @include mobile {
            width: 30vw;
            margin-top: 4%;
            margin-left: 35vw;
          }
        }
      }
      @include mobile {
        padding-left: 0;
      }
    }
  }
}
