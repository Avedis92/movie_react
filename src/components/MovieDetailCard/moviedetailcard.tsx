import { useState } from 'react';
import ModalVideo from 'react-modal-video';
import TrailerPlayer from '../TrailerPlayer/trailerplayer';
import styles from './moviedetailcard.module.scss';

function MovieDetailCard({
  backdropPath, posterPath, id, movieGenre, title, tagline, runtime, releaseDate, overview,
}:MovieDetails) {
  const [trailerKey, setTrailerKey] = useState('');
  const [isOpen, setIsOpen] = useState(false);
  const {
    container, InnerContainer, Infos, posterContainer, movieGenres, movieInfoContainer, movieHeaders, movieOverview, trailerWatch,
  } = styles;

  return (
    <>
      <div className={container} style={{ backgroundImage: `url(${backdropPath})` }}>
        {isOpen
          ? (
            <ModalVideo
              channel="youtube"
              isOpen={isOpen}
              videoId={trailerKey}
              onClose={() => setIsOpen(false)}
            />
          )
          : null}
        <div className={InnerContainer}>
          <div className={Infos}>
            <div className={posterContainer}>
              <img src={posterPath!} alt={title!} />
            </div>
            <div className={movieInfoContainer}>
              <div className={movieHeaders}>
                <h1>{title}</h1>
                {tagline ? <h2>{tagline}</h2> : null}
                <div className={movieGenres}>
                  {movieGenre.map((myMovieGenre:Genre) => (
                    <h2 key={myMovieGenre.id}>
                      {myMovieGenre.name}
                      {'  '}
                      {'  '}
                    </h2>
                  ))}
                </div>
                <p>
                  {runtime}
                  min
                </p>
                <p>
                  {releaseDate}
                </p>
              </div>
              <p className={movieOverview}>{overview}</p>
              <div className={trailerWatch}>
                <TrailerPlayer
                  movieId={id}
                  buttonType="watchNowButton"
                  openMovieTrailers={(key) => {
                    if (key !== '-1') {
                      setTrailerKey(key);
                      setIsOpen(true);
                    }
                  }}
                />
              </div>
            </div>
          </div>
        </div>
      </div>

    </>
  );
}
export default MovieDetailCard;
