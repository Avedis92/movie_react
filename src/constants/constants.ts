export const apiKey = '0600befd930f4be78f3d1123f393affc';
export const baseUrl = 'https://api.themoviedb.org/3/';
export const trendingMovieUrl = `${baseUrl}trending/movie/week?api_key=${apiKey}&page=`;
export const topRatedMovieUrl = `${baseUrl}movie/top_rated?api_key=${apiKey}&page=`;
export const upcomingMovieUrl = `${baseUrl}movie/upcoming?api_key=${apiKey}&page=`;
export const movieGenreListUrl = `${baseUrl}genre/movie/list?api_key=${apiKey}`;
export const searchMovieUrl = `${baseUrl}search/movie?api_key=${apiKey}&query=`;
export const postPathUrl = 'https://image.tmdb.org/t/p/w500';
export const defaultBackgroundImage = 'https://media-assets-05.thedrum.com/cache/images/thedrum-prod/s3-news-tmp-10557-netflixa--default--1280.jpg';
export const pageNotFoundBackgroundImage = 'https://cdn3.vectorstock.com/i/1000x1000/67/12/404-error-page-background-vector-20596712.jpg';
export const movieVideoUrl = `${baseUrl}/movie/`;
export const searchMovieByGenreUrl = `${baseUrl}discover/movie?api_key=${apiKey}&with_genres=`;
export const defaultPageNumber = 1;
export const defaultMovieType = 'trendings';
