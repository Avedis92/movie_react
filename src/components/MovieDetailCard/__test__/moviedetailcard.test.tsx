import { render } from '@testing-library/react';
import MovieDetailCard from '../moviedetailcard';
import { postPathUrl } from '../../../constants/constants';

test('Create a snapshot test', () => {
  const {asFragment} = render(<MovieDetailCard
    backdropPath={`${postPathUrl}/fCayJrkfRaCRCTh8GqN30f8oyQF.jpg`}
    posterPath={`${postPathUrl}/fCayJrkfRaCRCTh8GqN30f8oyQF.jpg`}
    id={550}
    movieGenre={[{ id: 18, name: 'Drama' }]}
    title="Fight Club"
    tagline={"How much can you know about yourself if you've never been in a fight?"}
    runtime={139}
    releaseDate="1999-10-12"
    overview={'A ticking-time-bomb insomniac and a slippery soap salesman channel primal male aggression into a shocking new form of therapy. Their concept catches on, with underground "fight clubs" forming in every town, until an eccentric gets in the way and ignites an out-of-control spiral toward oblivion.'}
  />);
  expect(asFragment()).toMatchSnapshot();
});
