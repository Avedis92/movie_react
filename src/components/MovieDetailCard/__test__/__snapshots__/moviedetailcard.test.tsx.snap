// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create a snapshot test 1`] = `
<DocumentFragment>
  <div
    class="container"
    style="background-image: url(https://image.tmdb.org/t/p/w500/fCayJrkfRaCRCTh8GqN30f8oyQF.jpg);"
  >
    <div
      class="InnerContainer"
    >
      <div
        class="Infos"
      >
        <div
          class="posterContainer"
        >
          <img
            alt="Fight Club"
            src="https://image.tmdb.org/t/p/w500/fCayJrkfRaCRCTh8GqN30f8oyQF.jpg"
          />
        </div>
        <div
          class="movieInfoContainer"
        >
          <div
            class="movieHeaders"
          >
            <h1>
              Fight Club
            </h1>
            <h2>
              How much can you know about yourself if you've never been in a fight?
            </h2>
            <div
              class="movieGenres"
            >
              <h2>
                Drama    
              </h2>
            </div>
            <p>
              139min
            </p>
            <p>
              1999-10-12
            </p>
          </div>
          <p
            class="movieOverview"
          >
            A ticking-time-bomb insomniac and a slippery soap salesman channel primal male aggression into a shocking new form of therapy. Their concept catches on, with underground "fight clubs" forming in every town, until an eccentric gets in the way and ignites an out-of-control spiral toward oblivion.
          </p>
          <div
            class="trailerWatch"
          >
            <div
              class="watchNowContainer"
            >
              <button
                id="550"
                name="watchNowButton"
                type="button"
              >
                Watch Now
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
